components/markets/MarketContextActionOutcomeSelector.tsx

```diff
@@ -0,0 +1,133 @@
+import { Listbox, Transition } from "@headlessui/react";
+import { FullMarketFragment } from "@zeitgeistpm/indexer";
+import { MarketOutcomeAssetId, getIndexOf } from "@zeitgeistpm/sdk-next";
+import Input from "components/ui/Input";
+import TruncatedText from "components/ui/TruncatedText";
+import * as fuzzysort from "fuzzysort";
+import { calcMarketColors } from "lib/util/color-calc";
+import { useEffect, useMemo, useState } from "react";
+import { BsArrowLeft } from "react-icons/bs";
+import { RiArrowDownSLine } from "react-icons/ri";
+
+export type MarketContextActionOutcomeSelectorProps = {
+  market: FullMarketFragment;
+  selected: MarketOutcomeAssetId;
+  options?: MarketOutcomeAssetId[];
+  onChange: (selected: MarketOutcomeAssetId) => void;
+};
+
+export const MarketContextActionOutcomeSelector = ({
+  market,
+  selected,
+  options,
+  onChange,
+}: MarketContextActionOutcomeSelectorProps) => {
+  const [open, setOpen] = useState(false);
+  const [search, setSearch] = useState<string | undefined>();
+
+  let optionsLength = useMemo(() => options?.length, [options]);
+
+  if (search && options) {
+    let opts = options.map((option) => {
+      const name = market.categories?.[getIndexOf(option)].name ?? "";
+      return { option, name };
+    });
+
+    options = fuzzysort
+      .go(search, opts, { key: "name" })
+      .map((result) => result.obj.option);
+  }
+
+  useEffect(() => {
```

Comment:
I think this is better because the state regarding if its open or not is set in multiple places, so would have to be called every time the state explicetly changes to !open. Also the `MarketContextActionOutcomeSelector` is rendered containing its search state even when its closed, so clearing search on unmount would also not work as expected.